src/taskpane/taskpane.ts

```diff
@@ -37,3 +42,65 @@ async function run() {
     OfficeHelpers.Utilities.log(error);
   }
 }
+
+async function runCfTests() {  
+  await Excel.run(async context => {
+    for (let i = 0; i < testFunctions.length; i++) {
+      const range = context.workbook.getSelectedRange();
+      const formula : string = testFunctions[i]
+      range.formulas = [[formula]];
+      await context.sync();
+      await sleep(2000);
+
+      // Check to if this is a streaming function
+      await readData(formula.indexOf("INCREMENT") > 0)
+    }
+  });
+  sendData(cfValues);
+}
+
+async function readData(isStreamingFunction: boolean) {
+  await Excel.run(async context => {
+
+    // if this is a streaming function, we want to capture two values so we can
+    // validate the function is indeed streaming
+    for (let i = 0; isStreamingFunction ?  i <= 1 : i < 1; i++)
+    {
+      const range = context.workbook.getSelectedRange();
+      range.load("values");
+      await context.sync();
+  
+      var data = {"cfValue": range.values[0][0]};
+      cfValues.push(data);
+    }
+  });
+}
+
+async function sendData(values)
+{
+  //make cfValues a json blob that we can pass in single request to test server
+  var json = JSON.stringify(values);  
+    
+  const Http = new XMLHttpRequest();
+  const url: string =`https://localhost:8080/`;
+  let dataUrl : string = url + "results/?data=" + encodeURIComponent(json);
+  Http.open("GET", dataUrl, true);  
+  Http.setRequestHeader('Content-type','application/json; charset=utf-8');
+  Http.send();
+}
+
+async function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+function isTtestServerStarted() {
```

Comment:
I would make this `async function isTestServerStarted(): Promise<boolean>` return a promise which is resolve(true) if the 200 is returned, and resolve(false) if not.  That way, the runTests() function can first call `if (await isTestServerStarted()) {` and run the tests if the server is started. The code is cleaner that way.